lib/ansible/modules/network/netscaler/netscaler_nitro_request.py

```diff
@@ -0,0 +1,899 @@
+        required: true
+
+    nitro_protocol:
+        choices: [ 'http', 'https' ]
+        default: http
+        description:
+            - Which protocol to use when accessing the nitro API objects.
+
+    validate_certs:
+        description:
+            - If C(no), SSL certificates will not be validated. This should only be used on personally controlled sites using self-signed certificates.
+        default: 'yes'
+
+    nitro_auth_token:
+        description:
+            - The authentication token provided by the C(mas_login) operation. It is required when issuing NITRO API calls through a MAS proxy.
+
+    resource:
+        description:
+            - The type of resource we are operating on.
+            - It is required for all I(operation) values except C(mas_login) and C(save_config).
+
+    name:
+        description:
+            - The name of the resource we are operating on.
+            - "It is required for the following I(operation) values: C(update), C(get), C(delete)."
+
+    attributes:
+        description:
+            - The attributes of the NITRO object we are operating on.
+            - "It is required for the following I(operation) values: C(add), C(update), C(action)."
+
+    args:
+        description:
+            - A dictionary which defines the key arguments by which we will select the NITRO object to operate on.
+            - "It is required for the following I(operation) values: C(get_by_args), C('delete_by_args')."
+
+    filter:
+        description:
+            - A dictionary which defines the filter with which to refine the NITRO objects returned by the C(get_filtered) I(operation).
+
+    operation:
+        description:
+            - Define the NITRO operation that we want to perform.
+        choices:
+            - add
+            - update
+            - get
+            - get_by_args
+            - get_filtered
+            - get_all
+            - delete
+            - delete_by_args
+            - count
+            - mas_login
+            - save_config
+            - action
+
+    expected_nitro_errorcode:
+        description:
+            - A list of numeric values that signify that the operation was successful.
+        default: [0]
+        required: true
+
+    action:
+        description:
+            - The action to perform when the I(operation) value is set to C(action).
+            - Some common values for this parameter are C(enable), C(disable), C(rename).
+
+    instance_ip:
+        description:
+            - The ip address of the target Netscaler instance when issuing a NITRO request through a MAS proxy.
+
+    instance_name:
+        description:
+            - The name of the target Netscaler instance when issuing a NITRO request through a MAS proxy.
+
+    instance_id:
+        description:
+            - The id of the target Netscaler instance when issuing a NITRO request through a MAS proxy.
+'''
+
+EXAMPLES = '''
+
+- name: Add a server
```

Comment:
Each example task should be a continuous block, please do not have blank lines between sections